src/containerapp/azext_containerapp/containerapp_java_decorator.py

```diff
@@ -0,0 +1,133 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import Any, Dict
+
+from azure.cli.command_modules.containerapp._utils import safe_get, safe_set
+from azure.cli.core.commands import AzCliCommand
+from azure.cli.core.azclierror import ValidationError, ResourceNotFoundError
+from azure.cli.command_modules.containerapp.base_resource import BaseResource
+from knack.log import get_logger
+
+from azure.cli.command_modules.containerapp._utils import _get_existing_secrets
+from knack.util import CLIError
+
+from ._models import JavaLoggerSetting
+
+from ._client_factory import handle_raw_exception, handle_non_404_status_code_exception
+
+logger = get_logger(__name__)
+
+
+class ContainerappJavaLoggerDecorator(BaseResource):
+
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.containerapp_def = None
+        try:
+            self.containerapp_def = self.client.show(cmd=cmd,
+                                                resource_group_name=raw_parameters.get("resource_group_name"),
+                                                name=raw_parameters.get("name"))
+        except Exception as e:
+            handle_non_404_status_code_exception(e)
+
+        if not self.containerapp_def:
+            raise ResourceNotFoundError("The containerapp '{}' does not exist".format(raw_parameters.get("name")))
+
+        _get_existing_secrets(cmd, raw_parameters.get("resource_group_name"), raw_parameters.get("name"), self.containerapp_def)
+
+    def validate_arguments(self):
+        if self.get_argument_all() is None and self.get_argument_logger_name() is None:
+            raise CLIError(
+                'Either of --logger-name/--all needs to be specified.')
+
+        if self.get_argument_all() is not None and self.get_argument_logger_name() is not None:
+            raise CLIError(
+                'Both --logger-name and --all cannot be specified together.')
+
+        if not safe_get(self.containerapp_def['properties'], "configuration", "runtime", "java", "javaAgent", "enabled", default=False):
+            raise ValidationError(
+                "The containerapp '{}' does not enable java agent, "
+                "please run `az containerapp update --name {} --resource-group {} --enable-java-agent true` to enable java agent".format(
+                    self.get_argument_name(), self.get_argument_name(), self.get_argument_resource_group_name()))
+
+    def show(self):
+        loggers = safe_get(self.containerapp_def['properties'], 'configuration', 'runtime', 'java', 'javaAgent',
+                           'logging', 'loggerSettings', default=[])
+        if self.get_argument_all() is None:
+            for logger in loggers:
```

Comment:
`logger` is used by `logger = get_logger(__name__)`
![image](https://github.com/Azure/azure-cli-extensions/assets/46143499/25959aae-ca02-4bca-8823-e0c172653da4)